compilation error with gcc 4.8.5, only at level -O0 (oh-zero)

CentOS 7, system gcc 4.8.5.  Compiles with -O1 and up.  UCX master @ cf49bb3609b2dc77a75da4bf16b2fe697e806d26

```
make[2]: Entering directory `/home/tc/src/ucx/git/build/src/uct'
  CC       base/libuct_la-uct_md.lo
...
  CC       ib/ud/base/libuct_la-ud_iface.lo
In file included from /home/tc/src/ucx/git/build/../source/src/ucs/datastruct/list.h:10:0,
                 from /home/tc/src/ucx/git/build/../source/src/uct/base/uct_worker.h:12,
                 from ../../../source/src/uct/ib/ud/base/ud_iface.h:11,
                 from ../../../source/src/uct/ib/ud/base/ud_iface.c:7:
../../../source/src/uct/ib/ud/base/ud_iface.c: In function ‘uct_ud_iface_calc_gid_len’:
../../../source/src/uct/ib/ud/base/ud_iface.c:392:63: error: case label does not reduce to an integer constant
     UCS_STATIC_ASSERT((UCT_IB_GRH_LEN - (20 + offsetof(struct iphdr, daddr))) == ipv4_len);
                                                               ^
/home/tc/src/ucx/git/build/../source/src/ucs/sys/compiler.h:31:30: note: in definition of macro ‘UCS_STATIC_ASSERT’
      switch(0) {case 0:case (_cond):;}
                              ^
../../../source/src/uct/ib/ud/base/ud_iface.c:395:56: error: case label does not reduce to an integer constant
     UCS_STATIC_ASSERT(UCT_IB_GRH_LEN - offsetof(struct ibv_grh, dgid) == ipv6_len);
                                                        ^
/home/tc/src/ucx/git/build/../source/src/ucs/sys/compiler.h:31:30: note: in definition of macro ‘UCS_STATIC_ASSERT’
      switch(0) {case 0:case (_cond):;}
                              ^
make[2]: *** [ib/ud/base/libuct_la-ud_iface.lo] Error 1
make[2]: Leaving directory `/home/tc/src/ucx/git/build/src/uct'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/tc/src/ucx/git/build'
make: *** [all] Error 2
```

--- a/contrib/test_jenkins.sh
+++ b/contrib/test_jenkins.sh
@@ -228,6 +228,17 @@ build_icc() {
 }
 
 #
+# Build debug version
+#
+build_debug() {
+	echo "==== Build with --enable-debug option ===="
+	../contrib/configure-devel --prefix=$ucx_inst --enable-debug
+	$MAKE clean
+	$MAKE
+	$MAKE distclean
+}
+
+#
 # Build with clang compiler
 #
 build_clang() {
@@ -546,6 +557,7 @@ run_tests() {
 	export UCX_ERROR_MAIL_FOOTER=$JOB_URL/$BUILD_NUMBER/console
 
 	do_distributed_task 0 4 build_icc
+	do_distributed_task 1 4 build_debug
 	do_distributed_task 3 4 build_clang
 
 	# all are running mpi tests

--- a/src/uct/ib/ud/base/ud_iface.c
+++ b/src/uct/ib/ud/base/ud_iface.c
@@ -17,6 +17,10 @@
 #include <linux/ip.h>
 
 
+#define UCT_UD_IPV4_ADDR_LEN sizeof(struct in_addr)
+#define UCT_UD_IPV6_ADDR_LEN sizeof(struct in6_addr)
+
+
 SGLIB_DEFINE_LIST_FUNCTIONS(uct_ud_iface_peer_t, uct_ud_iface_peer_cmp, next)
 SGLIB_DEFINE_HASHED_CONTAINER_FUNCTIONS(uct_ud_iface_peer_t,
                                         UCT_UD_HASH_SIZE,
@@ -384,24 +388,24 @@ void uct_ud_iface_remove_async_handlers(uct_ud_iface_t *iface)
  * buffer). In this case, the content of the first 20 bytes is undefined." */
 static void uct_ud_iface_calc_gid_len(uct_ud_iface_t *iface)
 {
-    const int ipv4_len      = sizeof(struct in_addr);
-    const int ipv6_len      = sizeof(struct in6_addr);
     uint16_t *local_gid_u16 = (uint16_t*)iface->super.gid.raw;
 
     /* Make sure that daddr in IPv4 resides in the last 4 bytes in GRH */
-    ucs_assert_always((UCT_IB_GRH_LEN - (20 + offsetof(struct iphdr, daddr))) == ipv4_len);
+    UCS_STATIC_ASSERT((UCT_IB_GRH_LEN - (20 + offsetof(struct iphdr, daddr))) ==
+                      UCT_UD_IPV4_ADDR_LEN);
 
     /* Make sure that dgid resides in the last 16 bytes in GRH */
-    ucs_assert_always((UCT_IB_GRH_LEN - offsetof(struct ibv_grh, dgid)) == ipv6_len);
+    UCS_STATIC_ASSERT((UCT_IB_GRH_LEN - offsetof(struct ibv_grh, dgid)) ==
+                      UCT_UD_IPV6_ADDR_LEN);
 
     /* IPv4 mapped to IPv6 looks like: 0000:0000:0000:0000:0000:ffff:????:????,
      * so check for leading zeroes and verify that 11-12 bytes are 0xff.
      * Otherwise either RoCEv1 or RoCEv2/IPv6 are used. */
     if (local_gid_u16[0] == 0x0000) {
         ucs_assert_always(local_gid_u16[5] == 0xffff);
-        iface->config.gid_len = ipv4_len;
+        iface->config.gid_len = UCT_UD_IPV4_ADDR_LEN;
     } else {
-        iface->config.gid_len = ipv6_len;
+        iface->config.gid_len = UCT_UD_IPV6_ADDR_LEN;
     }
 }
 

--- a/src/uct/ib/ud/base/ud_iface.c
+++ b/src/uct/ib/ud/base/ud_iface.c
@@ -389,10 +389,10 @@ static void uct_ud_iface_calc_gid_len(uct_ud_iface_t *iface)
     uint16_t *local_gid_u16 = (uint16_t*)iface->super.gid.raw;
 
     /* Make sure that daddr in IPv4 resides in the last 4 bytes in GRH */
-    UCS_STATIC_ASSERT((UCT_IB_GRH_LEN - (20 + offsetof(struct iphdr, daddr))) == ipv4_len);
+    ucs_assert_always((UCT_IB_GRH_LEN - (20 + offsetof(struct iphdr, daddr))) == ipv4_len);
 
     /* Make sure that dgid resides in the last 16 bytes in GRH */
-    UCS_STATIC_ASSERT(UCT_IB_GRH_LEN - offsetof(struct ibv_grh, dgid) == ipv6_len);
+    ucs_assert_always((UCT_IB_GRH_LEN - offsetof(struct ibv_grh, dgid)) == ipv6_len);
 
     /* IPv4 mapped to IPv6 looks like: 0000:0000:0000:0000:0000:ffff:????:????,
      * so check for leading zeroes and verify that 11-12 bytes are 0xff.

